src/AmpImage/__tests__/ApmImage.test.js

```diff
@@ -0,0 +1,167 @@
+  const expected = objectMother()
+  const { container } = render(
+    <AmpImage fallbackSrc={expected.fallbackSrc} alt={expected.alt} />
+  )
+  const img = container.firstChild
+
+  expect(img).toHaveAttribute('width', expect.stringContaining('400'))
+  expect(img).toHaveAttribute('height', expect.stringContaining('225'))
+})
+
+test('allows you to manually set width and height if there are no instances', () => {
+  const expected = objectMother()
+  const { container } = render(
+    <AmpImage
+      fallbackSrc={expected.fallbackSrc}
+      alt={expected.alt}
+      fallbackWidth="900"
+      fallbackHeight="500"
+    />
+  )
+  const img = container.firstChild
+
+  expect(img).toHaveAttribute('width', expect.stringContaining('900'))
+  expect(img).toHaveAttribute('height', expect.stringContaining('500'))
+})
+
+test('creates the correct AMP image when properly formatted image data is provided', () => {
+  const expected = objectMother()
+
+  const { container } = render(<AmpImage image={image} />)
+
+  // getByAltText doesn't work for amp-img? Maybe I can do a PR! :eyes:
+  const img = container.firstChild
+
+  expect(img).toBeInTheDocument()
+  expect(img).toHaveAttribute(
+    'srcSet',
+    expect.stringContaining(expected.srcSet)
+  )
+  expect(img).toHaveAttribute('src', expect.stringContaining(expected.src))
+})
+
+test('allows you to set the class with the elementClass property', () => {
+  const expected = objectMother()
+
+  const { container } = render(
+    <AmpImage
+      image={imageWithPreferred}
+      elementClass="test"
+      aspectRatio="widescreen"
+      sizes="(min-width: 960px) 720px, 100vw"
+    />
+  )
+
+  const img = container.firstChild
+  expect(img).toBeInTheDocument()
+  expect(img).toHaveAttribute('src', expect.stringContaining(expected.src))
+  expect(img).toHaveAttribute('class', expect.stringContaining(expected.class))
+})
+
+test('creates image when all fallbacks are provided', () => {
+  const expected = objectMother()
+
+  const { container } = render(
+    <AmpImage
+      fallbackSrc={expected.fallbackSrc}
+      fallbackSrcSet={expected.fallbackSrcSet}
+      alt={expected.alt}
+    />
+  )
+
+  const img = container.firstChild
+  expect(img).toBeInTheDocument()
+  expect(img).toHaveAttribute(
+    'src',
+    expect.stringContaining(expected.fallbackSrc)
+  )
+  expect(img).toHaveAttribute(
+    'srcSet',
+    expect.stringContaining(expected.fallbackSrcSet)
+  )
+  expect(img).toHaveAttribute('alt', expect.stringContaining(expected.alt))
+})
+
+//// FAILURES
+
+test('throws when provided poorly shaped image data', () => {
+  let badData = {
+    poop:
+      'https://s3-us-west-2.amazonaws.com/s.cdpn.io/298/wolf_20131015_003_1400.jpg 1400w, https://s3-us-west-2.amazonaws.com/s.cdpn.io/298/wolf_20131015_003_700.jpg 700w, https://s3-us-west-2.amazonaws.com/s.cdpn.io/298/wolf_20131015_003_400.jpg 400w',
+    alt: 4
+  }
+
+  let renderDidFail = false
+
+  try {
```

Comment:
Here the test is responsible for throwing an error should the component have this responsibility to throw instead?

Reply:
maybe the props types do this  not sure